### DISCUSSION TITLE
Why is the Nuid implementation internal?

### DISCUSSION THREAD

**Discussion by Madgvox** [nats-io/nats.net#616] [General]
I'd like to generate my own Nuids. Since I'm importing the library anyway, feels redundant to have to copy over the implementation to be able to accomplish this.

If there's not a specific reason, I'd like to propose the implementation be made public.

**Comment by mtmk** [comment]
> If there's not a specific reason, I'd like to propose the implementation be made public.

not really we can open it up. feel free to put up a PR ;) cc @jasper-d

**Reply by Madgvox** [reply]
Will do 👍

**Reply by Madgvox** [reply]
Done! https://github.com/nats-io/nats.net/pull/618
